packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts

```diff
@@ -660,8 +854,171 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
+        agentContext,
+        revocationStatusList.revRegDefId
+      )
+
+      if (!revocationRegistryDefinition) {
+        return {
+          registrationMetadata: {},
+          revocationStatusListMetadata: {},
+          revocationStatusListState: {
+            revocationStatusList,
+            state: 'failed',
+            reason: `Unable to resolve revocation registry definition '${revocationStatusList.revRegDefId}'`,
+          },
+        }
+      }
+
+      const { credentialDefinition } = await this.getCredentialDefinition(
+        agentContext,
+        revocationRegistryDefinition.credDefId
+      )
+
+      if (!credentialDefinition) {
+        return {
+          registrationMetadata: {},
+          revocationStatusListMetadata: {},
+          revocationStatusListState: {
+            revocationStatusList,
+            state: 'failed',
+            reason: `Unable to resolve credential definition '${revocationRegistryDefinition.credDefId}'`,
+          },
+        }
+      }
+
+      const { schemaMetadata } = await this.getSchema(agentContext, credentialDefinition.schemaId)
+
+      if (!schemaMetadata?.indyLedgerSeqNo || typeof schemaMetadata.indyLedgerSeqNo !== 'number') {
+        return {
+          registrationMetadata: {},
+          revocationStatusListMetadata: {},
+          revocationStatusListState: {
+            revocationStatusList,
+            state: 'failed',
+            reason: `Unable to resolve schema '${credentialDefinition.schemaId}'`,
+          },
+        }
+      }
+
+      const schemaSeqNo = Number(schemaMetadata.indyLegderSeqNo)
+      const didIndyRevocationRegistryEntryId = getDidIndyRevocationRegistryEntryId(
+        namespace,
+        namespaceIdentifier,
+        schemaSeqNo,
+        credentialDefinition.tag,
+        revocationRegistryDefinition.tag
+      )
+
+      const { endorsedTransaction, endorserDid, endorserMode } = options
+      if (endorsedTransaction) {
+        agentContext.config.logger.debug(
+          `Preparing endorsed tx '${endorsedTransaction}' for submission on ledger '${namespace}' with did '${revocationStatusList.issuerId}'`,
+          revocationStatusList
+        )
+        writeRequest = new CustomRequest({ customRequest: endorsedTransaction })
+        // extract the seqNo from the endorsed transaction, which is contained in the ref field of the operation
+      } else {
+        const revocationRegistryDefinitionRequest = new RevocationRegistryEntryRequest({
+          submitterDid: namespaceIdentifier,
+          revocationRegistryEntry: {
+            ver: '1.0',
+            value: {
+              accum: revocationStatusList.currentAccumulator,
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-expect-error
+              issued: revocationStatusList.issued ?? [],
+              revoked: revocationStatusList.revoked ?? [],
```

Comment:
So here we need to get the current state, and based on the current statusList and the new statusList we need to calculate the `issued` and `revoked` indices.